crates/client/rpc-core/src/utils.rs

```diff
@@ -157,6 +160,14 @@ pub fn to_declare_tx(tx: BroadcastedDeclareTransaction) -> Result<DeclareTransac
             let program: Program = Program::from_bytes(&decompressed_bytes, None)
                 .map_err(|_| anyhow!("Failed to deserialize the contract class program"))?;
 
+            let legacy_program = LegacyContractClass {
+                program: serde_json::from_slice(decompressed_bytes.as_slice())?,
+                abi: vec![],
```

Comment:
Cairo 0 class hashes depend on abi. You can't just supply an empty array.

`starknet-rs` supports converting from the RPC ABI entry type into the Cairo 0 ABI entry type directly. So you should just do this:

```suggestion
                abi: match declare_tx_v1.contract_class.abi.as_ref() {
                    Some(abi) => abi.iter().cloned().map(|entry| entry.into()).collect::<Vec<_>>(),
                    None => vec![],
                },
```

I've tested the suggested changes and the declare tx goes through without issues.